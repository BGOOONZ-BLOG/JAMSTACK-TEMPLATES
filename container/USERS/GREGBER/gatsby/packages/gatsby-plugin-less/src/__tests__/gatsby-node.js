describe(`gatsby-plugin-less`, () => {
  jest.mock(`../resolve`, () => module => `/resolved/path/${module}`)

  const actions = {
    setWebpackConfig: jest.fn(),
  }

  // loaders "mocks"
  const loaders = {
    miniCssExtract: () => `miniCssExtract`,
    css: args => `css(${JSON.stringify(args)})`,
    postcss: args => `postcss(${JSON.stringify(args)})`,
    null: () => `null`,
  }

  const { onCreateWebpackConfig } = require(`../gatsby-node`)

  beforeEach(() => {
    actions.setWebpackConfig.mockReset()
  })

  const tests = {
    stages: [`develop`, `build-javascript`, `develop-html`, `build-html`],
    options: {
      "No options": {},
      "Loader options #1": {
        loaderOptions: {
          appendData: `@env: ${process.env.NODE_ENV};`,
        },
      },
      "Less options #1": {
        lessOptions: {
          modifyVars: {
            "text-color": `#fff`,
          },
          strictMath: true,
        },
      },
      "Less options #2": {
        lessOptions: {
          modifyVars: require(`../theme-test.js`),
        },
      },
      "PostCss plugins": {
        postCssPlugins: [`test1`],
      },
      "css-loader options": {
        cssLoaderOptions: {
          camelCase: false,
        },
      },
    },
  }

  tests.stages.forEach(stage => {
    for (let label in tests.options) {
      const options = tests.options[label]
      it(`Stage: ${stage} / ${label}`, () => {
        onCreateWebpackConfig({ actions, loaders, stage }, options)
        expect(actions.setWebpackConfig).toMatchSnapshot()
      })
    }
  })
})
