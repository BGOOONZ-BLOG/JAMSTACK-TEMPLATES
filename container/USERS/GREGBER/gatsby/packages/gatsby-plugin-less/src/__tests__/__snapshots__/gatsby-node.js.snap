// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-less Stage: build-html / Less options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {
                          "modifyVars": Object {
                            "text-color": "#fff",
                          },
                          "strictMath": true,
                        },
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: build-html / Less options #2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {
                          "modifyVars": Object {
                            "text-color": "#fff",
                          },
                        },
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: build-html / Loader options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "appendData": "@env: test;",
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: build-html / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: build-html / PostCss plugins 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({\\"plugins\\":[\\"test1\\"]})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: build-html / css-loader options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "css({\\"camelCase\\":false,\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: build-javascript / Less options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {
                          "modifyVars": Object {
                            "text-color": "#fff",
                          },
                          "strictMath": true,
                        },
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {
                          "modifyVars": Object {
                            "text-color": "#fff",
                          },
                          "strictMath": true,
                        },
                        "sourceMap": false,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: build-javascript / Less options #2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {
                          "modifyVars": Object {
                            "text-color": "#fff",
                          },
                        },
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {
                          "modifyVars": Object {
                            "text-color": "#fff",
                          },
                        },
                        "sourceMap": false,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: build-javascript / Loader options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "appendData": "@env: test;",
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "appendData": "@env: test;",
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: build-javascript / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: build-javascript / PostCss plugins 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({\\"plugins\\":[\\"test1\\"]})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2})",
                    "postcss({\\"plugins\\":[\\"test1\\"]})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: build-javascript / css-loader options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"camelCase\\":false,\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"camelCase\\":false,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: develop / Less options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {
                          "modifyVars": Object {
                            "text-color": "#fff",
                          },
                          "strictMath": true,
                        },
                        "sourceMap": true,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {
                          "modifyVars": Object {
                            "text-color": "#fff",
                          },
                          "strictMath": true,
                        },
                        "sourceMap": true,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: develop / Less options #2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {
                          "modifyVars": Object {
                            "text-color": "#fff",
                          },
                        },
                        "sourceMap": true,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {
                          "modifyVars": Object {
                            "text-color": "#fff",
                          },
                        },
                        "sourceMap": true,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: develop / Loader options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "appendData": "@env: test;",
                        "lessOptions": Object {},
                        "sourceMap": true,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "appendData": "@env: test;",
                        "lessOptions": Object {},
                        "sourceMap": true,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: develop / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": true,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": true,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: develop / PostCss plugins 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({\\"plugins\\":[\\"test1\\"]})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": true,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2})",
                    "postcss({\\"plugins\\":[\\"test1\\"]})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": true,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: develop / css-loader options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"camelCase\\":false,\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": true,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"camelCase\\":false,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": true,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: develop-html / Less options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {
                          "modifyVars": Object {
                            "text-color": "#fff",
                          },
                          "strictMath": true,
                        },
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: develop-html / Less options #2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {
                          "modifyVars": Object {
                            "text-color": "#fff",
                          },
                        },
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: develop-html / Loader options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "appendData": "@env: test;",
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: develop-html / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: develop-html / PostCss plugins 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "css({\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({\\"plugins\\":[\\"test1\\"]})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-less Stage: develop-html / css-loader options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.less\\$/,
                  "use": Array [
                    "css({\\"camelCase\\":false,\\"modules\\":true,\\"importLoaders\\":2})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/less-loader",
                      "options": Object {
                        "lessOptions": Object {},
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.less\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
